ethcore/src/blockchain/blockchain.rs

```diff
@@ -1005,29 +1006,50 @@ impl BlockChain {
 			batch.extend_with_cache(db::COL_EXTRA, &mut *write_receipts, update.block_receipts, CacheUpdatePolicy::Remove);
 		}
 
-		{
-			let mut write_blocks_blooms = self.blocks_blooms.write();
-			batch.extend_with_cache(db::COL_EXTRA, &mut *write_blocks_blooms, update.blocks_blooms, CacheUpdatePolicy::Remove);
-		}
-
 		// These cached values must be updated last with all four locks taken to avoid
 		// cache decoherence
 		{
 			let mut best_block = self.pending_best_block.write();
+			let mut write_blocks_blooms = self.blocks_blooms.write();
```

Comment:
Lock order is changed and does not match the order of fields initialization.

Reply:
like in many other places in this file (although I check and locks never interfere) :) But you are right, I will bring back the old lock order, cause this change is not justified.